src/olympia/blocklist/tasks.py

```diff
@@ -88,28 +97,62 @@ def monitor_remote_settings():
+    filter_list: List[BlockType] = (
+        [] if filter_list is None else [BlockType[filter] for filter in filter_list]
+    )
     bucket = settings.REMOTE_SETTINGS_WRITER_BUCKET
     server = RemoteSettings(
         bucket, REMOTE_SETTINGS_COLLECTION_MLBF, sign_off_needed=False
     )
     mlbf = MLBF.load_from_storage(generation_time, error_on_missing=True)
+    is_base = len(filter_list) > 0
+    oldest_base_filter_id: int | None = None
+
     if is_base:
-        # clear the collection for the base - we want to be the only filter
-        server.delete_all_records()
-        statsd.incr('blocklist.tasks.upload_filter.reset_collection')
-        # Then the bloomfilter
-        data = {
-            'key_format': MLBF.KEY_FORMAT,
-            'generation_time': generation_time,
-            'attachment_type': BLOCKLIST_RECORD_MLBF_BASE,
-        }
-        with mlbf.storage.open(mlbf.filter_path, 'rb') as filter_file:
-            attachment = ('filter.bin', filter_file, 'application/octet-stream')
-            server.publish_attachment(data, attachment)
-            statsd.incr('blocklist.tasks.upload_filter.upload_mlbf')
+        for block_type in filter_list:
+            data = {
+                'key_format': MLBF.KEY_FORMAT,
+                'generation_time': generation_time,
+                'attachment_type': BLOCKLIST_RECORD_MLBF_BASE(block_type),
+            }
+            with mlbf.storage.open(mlbf.filter_path(block_type), 'rb') as filter_file:
+                attachment = ('filter.bin', filter_file, 'application/octet-stream')
+                server.publish_attachment(data, attachment)
+            base_filter_id: int | None = get_config(
+                MLBF_BASE_ID_CONFIG_KEY(block_type), int_value=True
+            )
+            if base_filter_id is not None:
+                if oldest_base_filter_id is None:
+                    oldest_base_filter_id = base_filter_id
+                else:
+                    oldest_base_filter_id = min(oldest_base_filter_id, base_filter_id)
+
+        statsd.incr('blocklist.tasks.upload_filter.upload_mlbf')
         statsd.incr('blocklist.tasks.upload_filter.upload_mlbf.base')
-    else:
+
+        if oldest_base_filter_id is not None:
+            for record in server.records():
+                record_time = int(
+                    record['stash_time']
+                    if 'stash_time' in record
+                    else record['generation_time']
+                    if 'generation_time' in record
+                    # If we don't have a generation time to use
+                    # then setting the base filter id will ensure
+                    # the record is not deleted.
+                    else oldest_base_filter_id
+                )
+                if record_time < oldest_base_filter_id:
+                    server.delete_record(record['id'])
+
+            cleanup_old_files.delay(base_filter_id=oldest_base_filter_id)
```

Comment:
Previous filters should also be deleted, and from what I can see locally, this isn't the case. We should only have a single record per filter.

Reply:
I'm not sure what you mean. This task is for cleaning up the local files and it only deletes them if they are both older than the given id and older than six months old. We can have multiple records per filter a `cache.json` a `stash.json` and or two filter files.

Reply:
ah sorry, I meant to comment where we do `server.delete_record()`.